src/main/java/org/radarcns/management/service/SubjectService.java

```diff
@@ -205,6 +206,19 @@ public SubjectDTO updateSubject(SubjectDTO newSubjectDto) {
                 .map(subjectMapper::subjectToSubjectReducedProjectDTO);
     }
 
+    /**
+     * Get all subjects created between dates specified.
+     *
+     * @param fromDate date created from
+     * @param toDate date created to
+     * @return the requested list of subjects
+     */
+    public List<SubjectDTO> findByDateCreated(ZonedDateTime fromDate, ZonedDateTime toDate) {
+        return this.findAll(new PageRequest(0,Integer.MAX_VALUE)).getContent().stream()
+                .filter(subject -> subject.getCreatedDate().isAfter(fromDate)
+                && subject.getCreatedDate().isBefore(toDate)).collect(Collectors.toList());
+    }
```

Comment:
This should take a `Pageable` and return a `Page` instead of a list. Right now, the database query does not actually limit the dates, pages or authorization.

1. query should take pageable parameter and return a page.
2. query should include date parameters instead of using findAll (this needs an override in `SubjectRepository`)
3. query should include authorization parameters instead of filtering afterwards, otherwise the pagination will be off.

The following conditions are needed for authorization:
- before the request `token.checkPermission(SUBJECT_READ)` must be called.
- if the token is with client credentials OR the user is SYS_ADMIN, no limits on users are imposed.

In those cases, no filtering is needed. We could say we want to limit this API to OAuth clients and admins, in that case, we're done. All other types should get a forbidden.

Otherwise, you need to enumerate all projects where the token has role `PROJECT_*`. You can then create a query like `WHERE ... AND s.project.projectName IN ({$projectNames}). You would have to make a method for that in `SubjectRepository`.